core/src/main/java/com/google/instrumentation/trace/Samplers.java

```diff
@@ -98,6 +102,7 @@ public String toString() {
     }
   }
 
+  @Immutable
   private static final class ProbabilitySampler extends Sampler {
```

Comment:
Gave a try and it looks decent. So I will keep it for the moment, even if I feel that the class is not a value class.